### ISSUE TITLE
[BUG] Wrong item being selected on scroll

### ISSUE THREAD

**Issue by shyamal890** [d3lm/ngx-drag-to-select#63]
Checkout this example: https://stackblitz.com/edit/ngx-drag-to-select-scroll-error?file=src/app/core/ui/components/layout/layout.component.html

Scroll a little and try selecting an item in the 2nd row.

**Comment by d3lm** [comment]
Thanks for the issue. This is not a bug actually. The documentation tells you how to deal with nested scrollable containers. The library only accounts for the container and / or the body as a scrollable container. If any other element is used to be the scrollable container you have to call `update` on the container component because the bounding boxes are not re-calculated.

Check out the [readme](https://github.com/d3lm/ngx-drag-to-select#how-do-i-deal-with-nested-scrollable-containers). It describes how to deal with nested scrollable containers.

The library shouldn't look for any other scrollable containers other than the container component and the body itself. Everything else is too app-specific.

**Comment by shyamal890** [comment]
One simply cannot update the container on every scroll update right?! It wouldn't even work with `drag to select` as the drawing box would flicker.

**Comment by d3lm** [comment]
I think I cannot follow, do you mind elaborating? Also I am very open for alternatives if you have one.

The problem is that the selection is based on the client bounding rect of each selectable element and if you scroll, that bounding box changes. As I said, the lib listens for scroll events on the select-container and the body and takes those into account but no other elements as this would be very app-specific. IMHO the developer has to take care of this themselves, of course this can be dealt with in a more efficient way, for example by auditing or throttling the scroll event on your side.

I personally have no idea how to deal with this in a different way and again, I think this is something very specific if you have other scrollable containers. We had the same discussion in an other issue #36 . What do you think @timdeschryver?

If you have an alternative, please go ahead and propose one.

**Comment by shyamal890** [comment]
It would be great to have something like `listenToClass` where the user can point to the container which the lib has to listen to for `size changes` or `scroll` events.

**Comment by shyamal890** [comment]
Hey @d3lm when route changes and the list is refreshed, even after utilizing the `update` function the library somehow multiple selects items when only one item was clicked.

I am also seeing wrong item being selected on route change + refresh.

I would try and reproduce this issue in stackblitz but seems like stackblitz is down for now.

**Comment by d3lm** [comment]
Please open a new issue for this and please provide a **minimal** reproducable.

**Comment by timdeschryver** [comment]
Passing an element seems like a good option to me, or would a directive be better here?
For what it is worth, if I remember correctly, we didn't want to introduce it previously because you could have nested scrollbars.

> One simply cannot update the container on every scroll update right?! 
Did you test the flickering @shyamal890 ?

**Comment by shyamal890** [comment]
@timdeschryver I did created another [issue](https://github.com/d3lm/ngx-drag-to-select/issues/67) which had basically 2 points one that the element was not selected when using `[selectOnDrag]="false"` and another one being that the drag box was causing issues on view change, which I am still facing.

**Comment by d3lm** [comment]
@timdeschryver Yes, that's exactly why we didn't introduce that. I like the idea of a directive, because you could potentially register multiple scroll containers. That sounds reasonable.

**Comment by d3lm** [comment]
@timdeschryver I think a directive is not really sufficient enough because there might be scroll containers outside of the `select-container` and I wouldn't be able to query those using the `@ContentChildren` decorator. So maybe the user can just pass in a list of element refs and is responsible for collecting all those elements and pass them along?

**Comment by timdeschryver** [comment]
@d3lm I could be completely wrong with this, but won't they all be children of `dts-select-container`?


I think to give the responsibility to the user of passing elements refs is fine. People using UI libraries like angular material for example already are used to this, so doing this would also feel comfortable.

**Comment by d3lm** [comment]
> [...] but won't they all be children of `dts-select-container`?

Not necessarily, because you could also have scroll containers outside of the select-container no? For instance in many apps the `body` is not scrollable but instead some `main` element that wraps the content area. Really depends on the layout tho.

The fact with Angular Material is a good one. So maybe we just off-load the responsibility to the user. However, we could still provide a directive that the user can use to mark elements as `dts-scroll-container` maybe? And then the user could query those and pass them to the container?

**Comment by alexbazile** [comment]
I did some kind of implementation to listen to the scroll event and to call update the library check it out here: 

https://stackblitz.com/edit/ngx-drag-to-select-scroll-error-hrn8de
